libtenzir/builtins/operators/to_opensearch.cpp

```diff
@@ -0,0 +1,519 @@
+      return;
+    }
+    const auto [ec, http_code] = req.get<curl::easy::info::response_code>();
+    TENZIR_ASSERT(ec == curl::easy::code::ok);
+    if (http_code < 200 or http_code > 299) {
+      diagnostic::warning("issue sending data. HTTP response code `{}`",
+                          http_code)
+        .note("response body: {}", response)
+        .primary(args_.operator_location)
+        .emit(dh);
+      return;
+    }
+    auto json = from_json(response);
+    if (not json.has_value()) {
+      return;
+    }
+    const auto r = try_as<record>(json.value());
+    if (not r) {
+      return;
+    }
+    const auto it = r->find("errors");
+    if (it == r->end()) {
+      return;
+    }
+    const auto errors = as<bool>(it->second);
+    if (errors) {
+      diagnostic::warning("issue sending data")
+        .note("response body: {}", response)
+        .primary(args_.operator_location)
+        .emit(dh);
+    }
+  }
+
+  auto
+  operator()(generator<table_slice> input,
+             operator_control_plane& ctrl) const -> generator<std::monostate> {
+    auto& dh = ctrl.diagnostics();
+    auto wrapped_req = new_req(dh);
+    if (not wrapped_req) {
+      co_return;
+    }
+    auto req = std::move(wrapped_req).unwrap();
+    auto b = json_builder{
+      {
+        .style = no_style(),
+        .oneline = true,
+        .omit_nulls = not args_.include_nulls,
+        .omit_empty_records = false,
+        .omit_empty_lists = false,
+      },
+      args_.max_content_length->inner,
+      args_.compress.has_value(),
+    };
+    for (auto last = time::clock::now(); auto&& slice : input) {
+      const auto now = time::clock::now();
+      if (now - last > args_.buffer_timeout->inner and b.has_contents()) {
+        send_req(req, b.yield(dh), dh);
+        last = now;
+      }
+      if (slice.rows() == 0) {
+        co_yield {};
+        continue;
+      }
+      // XXX: Why?
+      slice = resolve_enumerations(std::move(slice));
+      constexpr auto ng = []() -> generator<std::optional<std::string_view>> {
+        co_yield {std::nullopt};
+      };
+      const auto ids = resolve_str("id", args_.id, slice, dh);
+      const auto idxs = resolve_str("index", args_.index, slice, dh);
+      const auto acts = resolve_str("action", args_.action, slice, dh);
+      const auto docs = eval(
+        args_.doc.value_or(ast::this_{args_.operator_location}), slice, dh);
+      constexpr auto ty = string_type{};
+      auto id = ids ? values(ty, as<arrow::StringArray>(*ids->array)) : ng();
+      auto idx = idxs ? values(ty, as<arrow::StringArray>(*idxs->array)) : ng();
+      auto act = acts ? values(ty, as<arrow::StringArray>(*acts->array)) : ng();
+      auto doc = docs.values();
+      for (auto&& doc : docs.values()) {
+        const auto ptr = try_as<view<record>>(doc);
+        if (not ptr) {
+          diagnostic::warning("`doc` evaluated to non-record, skipping event")
+            .primary(*args_.doc)
+            .emit(dh);
```

Comment:
This needs to advance all the other generators. The best solution would be to advance/get all of them at the top of the loop.